### PR TITLE
Update transform feedback to use effective size

### LINKED ISSUE #10640: Confirm new effective size for transform feedback is correct
From https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/7642

@ziga-lunarg can you confirm `02361`/`02362`/`02363` are still correct and we are not now missing a different edge case

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Closes #10640

### CODE REVIEW

File: tests/unit/transform_feedback.cpp [hunk 1]
```diff
@@ -165,7 +165,7 @@ TEST_F(NegativeTransformFeedback, CmdBindTransformFeedbackBuffersEXT) {
             VkDeviceSize const offsets[1]{};
             VkDeviceSize const sizes[1]{tf_properties.maxTransformFeedbackBufferSize + 1};
 
-            m_errorMonitor->SetDesiredError("VUID-vkCmdBindTransformFeedbackBuffersEXT-pSize-02361");
```
Comment (spencer-lunarg) [inline tests/unit/transform_feedback.cpp:168]: so wait, is there no way to hit `02361` or `02362` now?

Comment (ziga-lunarg) [reply tests/unit/transform_feedback.cpp:168]: Yes there is, but 02363 covers it, so it should be removed
